src/test/util/setup_common.h

```diff
@@ -24,6 +24,9 @@
 /** This is connected to the logger. Can be used to redirect logs to any other log */
 extern const std::function<void(const std::string&)> G_TEST_LOG_FUN;
 
+/** Retrieve the command line arguments. */
+extern const std::function<std::vector<const char*>()> G_TEST_COMMAND_LINE_ARGUMENTS;
```

Comment:
c79eb4ad headers where std::functional is added
```diff
+++ src/test/util/setup_common.h
+#include <functional>
 #include <type_traits>
 #include <vector>
```
```diff
+++ src/test/main.cpp 
 #include <test/util/setup_common.h>
 
+#include <functional>
 #include <iostream>
```
...and so on.

Reply:
Added `#include <functional>` in all places where `std::function` is used (relevant to this PR).